### PR TITLE
feat(python): .to_numpy(use_pyarrow=False) for Object and Boolean

### PR DESCRIPTION
**Pull request by jonashaag**
Fixes https://github.com/pola-rs/polars/issues/7689
Fixes https://github.com/pola-rs/polars/issues/8393 (partly)

This disables the Arrow route with a panic because on the Arrow route objects are converted to binary strings for some reasons I don't understand (`object_series_to_arrow_array`)

### CODE REVIEW

File: py-polars/src/series.rs [hunk 1]
```diff
@@ -822,6 +822,9 @@ impl PySeries {
 
     pub fn to_arrow(&mut self) -> PyResult<PyObject> {
         self.rechunk(true);
+        if matches!(self.series.dtype(), DataType::Object(_)) {
```
Comment (ritchie46) [inline py-polars/src/series.rs:825]: Pyarrow doesn't support objects, but we use an extension array type for objects that is supported by arrow. This support has utility to go round trip via polars object to arrow and back.

This panic is a regression and should be removed.

**Review by ritchie46 — COMMENTED** [review]
Thanks @jonashaag. Looks great, except for the introduced panic. See the comments there.

File: py-polars/src/series.rs (hunk 1 above)
Comment (jonashaag) [reply py-polars/src/series.rs:825]: Do you want me to just remove this panic and leave the old like it was before?

Comment (ritchie46) [reply py-polars/src/series.rs:825]: Yes. :)

**Pull request merged**